### ISSUE TITLE
Fix issues with GCC-6

### ISSUE THREAD

**Issue by eribertomota** [sshock/AFFLIBv3#9]
Hi Phillip,

The attached patches were used in last Debian revision to fix some issues with GCC-6.

Thanks in advance.

Regards,

Eriberto

[30_fix_gcc_warnings_literal-suffix.patch.zip](https://github.com/sshock/AFFLIBv3/files/258446/30_fix_gcc_warnings_literal-suffix.patch.zip)
[40_fix_gcc_warnings_misleading-indentation.patch.zip](https://github.com/sshock/AFFLIBv3/files/258447/40_fix_gcc_warnings_misleading-indentation.patch.zip)

**Comment by sshock** [comment]
Fixed in 0f67bdd9ec446e5be791c3d69ce605d774156c94 and 4fd23a9bcade37e69c52eb32c167e785536c4dc0 with some caveats:

1. When fixing string literals like PRIu32, I added a space after as well, not just before.

2. I don't believe the change to String.h and Vector.h was correct.  I think the brackets added cause incorrect behavior in those functions.  So instead of adding brackets, I just fixed the indent.
